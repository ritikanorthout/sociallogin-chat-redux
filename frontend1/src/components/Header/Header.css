.logouButton {
  background-color: white;
  color: black;
  border: 1px solid rgb(7, 7, 7);
  cursor: pointer;
  position: inherit;
  margin-left: 1200px;
  top: 30px;
  right: 30px;
}
