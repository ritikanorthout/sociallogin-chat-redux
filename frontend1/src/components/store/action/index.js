import { AUTH_LOGIN, USER_REGISTER } from "../../constant/type";

// export const fngetLogin = (data) => {
//   return (dispatch) => {
//     dispatch({
//       type: AUTH_LOGIN,
//       payload: data,
//     });
//   };
// };

// export const fnuserRegister = (userData) => {
//   return (dispatch) => {
//     dispatch({
//       type: USER_REGISTER,
//       payload: userData,
//     });
//   };
// };

export const getLogin = (data) => {
  return (dispatch) => {
    dispatch({
      type: AUTH_LOGIN,
      payload: data,
    });
  };
};

export const userRegister = (userData) => {
  return (dispatch) => {
    dispatch({
      type: USER_REGISTER,
      payload: userData,
    });
  };
};
