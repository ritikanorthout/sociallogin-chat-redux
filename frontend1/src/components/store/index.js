import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import Reducer from "./reducer";

function saveToLocalStorage(state) {

  console.log('saveToLocalStorage ', state);

  try {
    const serializedState = JSON.stringify(state);
    localStorage.setItem("state", serializedState);
  } catch (error) {
    console.log(error);
  }
}

function loadToStore() {

  console.log('loadToStore');
  try {
    const serializedState = localStorage.getItem("state");
    console.log('serializedState',serializedState);
    if (serializedState === null) return undefined;
    return JSON.parse(serializedState);
  } catch (error) {
    console.log(error);
  }
}
const per = loadToStore();
const store = createStore(Reducer, per, applyMiddleware(thunk));

store.subscribe(() => saveToLocalStorage(store.getState()));
export default store;
