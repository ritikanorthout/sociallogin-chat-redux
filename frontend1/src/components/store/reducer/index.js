import { AUTH_LOGIN, USER_REGISTER, LOGOUT } from "../../constant/type";
const initialState = {
  isLoggedin: false,
  userInfo: { name: "", email: "", mobile: "", token: "", imgUrl: "" },
  userList: [],
};

const Reducer = (state = initialState, action) => {
  console.log("action.payload : ",action.payload);

  console.log("state : ",state);

  switch (action.type) {
    case AUTH_LOGIN:
      return {
        ...state,
        isLoggedin: true,
        userInfo: {
          name: action.payload.name,
          email: action.payload.email,
          imgUrl: action.payload.imageUrl,
        },
      };
    case USER_REGISTER:
      return {
        ...state,
        userList: [...state.userList, action.payload],
      };
    case LOGOUT:
      return {
        ...state,
        userInfo: { name: "", email: "", mobile: "", token: "", imgUrl: "" },
        isLoggedin: false,
      };
    default:
      return {
        ...state,
      };
  }

};

export default Reducer;
