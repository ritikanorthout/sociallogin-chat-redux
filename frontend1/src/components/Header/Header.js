import React from "react";
// import  from "react-router-dom";
import "./Header.css";
import { useDispatch } from "react-redux";

import { LOGOUT } from "../../components/constant/type";
export default function Header() {
  const dispatch = useDispatch();

  const Logout = () => {
    dispatch({ type: LOGOUT });
  };

  return (
    <div>
      <div className="callout primary" id="Header">
        <div className="row column">
          <button className="logouButton" onClick={Logout}>
            logout
          </button>
        </div>
      </div>
    </div>
  );
}
