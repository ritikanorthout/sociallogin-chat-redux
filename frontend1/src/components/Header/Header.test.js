import React from "react";
import ReactDOM from "react-dom";
import Header from "./Header";
import { Provider } from "react-redux";
import store from "../store";

it("Header renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <Provider store={store}>
      <Header />
    </Provider>,
    div
  );

  expect(div).toMatchSnapshot();
});
