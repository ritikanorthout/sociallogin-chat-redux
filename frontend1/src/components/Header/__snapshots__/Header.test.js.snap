// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without crashing 1`] = `
<div>
  <div>
    <div
      class="callout primary"
      id="Header"
    >
      <div
        class="row column"
      >
        <button
          class="logouButton"
        >
          logout
           
        </button>
      </div>
    </div>
  </div>
</div>
`;
